src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +32,47 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    this.tabs = _.reduce(tabsTemplate, function (current, tab) {
+      var id = tab.id.replace(":appid", appid);
+      if (activeTabId == null) {
+        activeTabId = id;
+      }
+      current.push({
+        id: id,
+        text: tab.text
+      });
+      return current;
```

Comment:
Can we get a white space before the `return`-statement? This is a style thing we do to make it more readable.